### PR TITLE
Add cont, spv, change return of /local, /poll, /listen

### LINKED ISSUE #9: Add cont requests, spv requests.
To allow cross-chain-transfers, add following functions: 

- Function that makes a request to `/spv` endpoint to fetch the proof. 
- Function that creates/makes cont request.

### PR DESCRIPTION
**Pull request by ggobugi27**
Add cont and spv, change return of /local, /poll, /listen
Closes #9

### CODE REVIEW

File: pact-lang-api-global.min.js [hunk 1]
```diff
@@ -5229,6 +5229,48 @@ var prepareExecCmd = function(keyPairs, nonce=new Date().toISOString(), pactCode
   return mkSingleCmd(sigs, cmd);
 };
 
+/**
+ * Prepare an ContMsg pact command for use in send or local execution.
+ * To use in send, wrap result with 'mkSingleCommand'.
+ * @param keyPairs {array or object} - array or single ED25519 keypair
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]: this can be "and/or clist" too

File: pact-lang-api-global.min.js [hunk 2]
```diff
@@ -5229,6 +5229,48 @@ var prepareExecCmd = function(keyPairs, nonce=new Date().toISOString(), pactCode
...
+ * @param nonce {string} - nonce value for ensuring unique hash - default to current time
+ * @param step {number} - the step on which pacts the command is sent to - required
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * @param step {number} - integer index of step to execute in defpact body - required
```

File: pact-lang-api-global.min.js [hunk 3]
```diff
@@ -5229,6 +5229,48 @@ var prepareExecCmd = function(keyPairs, nonce=new Date().toISOString(), pactCode
...
+ * @param proof {string} - JSON message of SPV proof - not required
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * @param proof {string} - JSON of SPV proof, required for cross-chain transfer. See `fetchSPV` below
```

File: pact-lang-api-global.min.js [hunk 4]
```diff
@@ -5229,6 +5229,48 @@ var prepareExecCmd = function(keyPairs, nonce=new Date().toISOString(), pactCode
...
+ * @param rollback {bool} - Bool of cont message - required
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * @param rollback {bool} - Indicates if this continuation is a rollback/cancel - required
```

File: pact-lang-api-global.min.js [hunk 5]
```diff
@@ -5229,6 +5229,48 @@ var prepareExecCmd = function(keyPairs, nonce=new Date().toISOString(), pactCode
...
+ * @param pactId {string} - pactId of the cont Msg - required
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * @param pactId {string} - identifies the already-begun Pact execution that this is continuing - required
```

File: pact-lang-api-global.min.js [hunk 6]
```diff
@@ -5196,7 +5196,7 @@ var pullAndCheckHashs = function(sigs) {
 /**
  * Prepare an ExecMsg pact command for use in send or local execution.
  * To use in send, wrap result with 'mkSingleCommand'.
- * @param keyPairs {array or object} - array or single ED25519 keypair and/or clist
+ * @param keyPairs {array or object} - array or single ED25519 keypair and/or clist (list of `cap` in mkCap)
  * @param nonce {string} - nonce value for ensuring unique hash
  * @param pactCode {string} - pact code to execute
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]: note this should indicate default like cont does below

File: pact-lang-api-global.min.js [hunk 7]
```diff
@@ -5422,31 +5471,78 @@ var mkReq = function(cmd) {
...
+  * @property type {string} - type of command - "cont" or "exec", default to "exec"
+  * @property pactCode {string} - pact code to execute in "exec" command - required for "exec"
+  * @property nonce {string} - nonce value to ensure unique hash - default to current time
+  * @property envData {object} - JSON message data for command - not required
+  * @property meta {object} - public meta information, see mkMeta
+  * @property networkId {object} network identifier of where the cmd is executed.
+  */
+
+ /**
+  * A contCmd to Execute in send
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js:5488]: unify these docs with changes above

File: pact-lang-api-global.min.js [hunk 8]
```diff
@@ -5422,31 +5471,78 @@ var mkReq = function(cmd) {
...
- * @property networkId {object} network identifier of where the cmd is executed.
- */
+
+
+ /**
+  * A execCmd Object to Execute in send or local.
+  * @typedef {Object} cmd to `/send` endpoint
+  * @property type {string} - type of command - "cont" or "exec", default to "exec"
+  * @property pactCode {string} - pact code to execute in "exec" command - required for "exec"
+  * @property nonce {string} - nonce value to ensure unique hash - default to current time
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js:5481]: note these docs are correct

File: pact-lang-api-global.min.js [hunk 9]
```diff
@@ -5455,15 +5551,15 @@ const fetchSend = async function(sendCmd, apiHost){
  */
 const fetchLocal = async function(localCmd, apiHost) {
   if (!apiHost)  throw new Error(`Pact.fetch.local(): No apiHost provided`);
-  const {keyPairs, nonce, pactCode, envData, meta, networkId} = localCmd
-  const cmd = prepareExecCmd(keyPairs, nonce, pactCode, envData, meta, networkId);
+  const {keyPairs, nonce, pactCode, envData, meta} = localCmd
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]: why no networkId?

File: pact-lang-api-global.min.js [hunk 10]
```diff
@@ -5455,15 +5551,15 @@ const fetchSend = async function(sendCmd, apiHost){
...
+  const cmd = prepareExecCmd(keyPairs, nonce, pactCode, envData, meta);
   const txRes = await fetch(`${apiHost}/api/v1/local`, mkReq(cmd));
   const tx = await txRes.json();
-  return tx.result;
+  return tx;
 };
 
 /**
- * Request poll Pact command to a running Pact server and retrieves tx result.
+ * Poll result of Pact command to a running Pact server and retrieves tx result.
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * Poll result of Pact command on a Pact server and retrieve tx result.
```

File: pact-lang-api-global.min.js [hunk 11]
```diff
@@ -5472,13 +5568,11 @@ const fetchPoll = async function(pollCmd, apiHost) {
...
   const resJSON = await res.json();
-  return Object.values(resJSON).map(res => {
-    return { reqKey: res.reqKey, result: res.result };
-  });
+  return resJSON;
 };
 
 /**
- * Request listen Pact command to a running Pact server and retrieves tx result.
+ * Listen result of Pact command to a running Pact server and retrieves tx result.
```
Comment (sirlensalot) [inline pact-lang-api-global.min.js]:
```suggestion
 * Listen for result of Pact command on a Pact server and retrieve tx result.
```

File: readme.md [hunk 12]
```diff
@@ -81,12 +81,17 @@ e.g. `JSON.stringify(.0000001) -> '1e-7'` is incorrect as Pact has infinite prec
 Simple fetch functions to make API request to a running Pact Server and retrieve the results.
 
 ```
-* A Command Object to Execute in send or local.
+* A "exec" or "cont" Command Object to Execute in send or local.
```
Comment (sirlensalot) [inline readme.md]: ensure these match comments above

**Review by sirlensalot — COMMENTED** [review]
My comments should be on `pact-lang-api.js`, not the global.min.js, sorry

File: pact-lang-api-global.min.js (hunk 9 above)
Comment (ggobugi27) [reply pact-lang-api-global.min.js]: I noticed that /local calls don't need networkId

**Pull request merged**